src/v1/device_manager_client.ts

```diff
@@ -1957,6 +1991,20 @@ export class DeviceManagerClient {
               res.on('end', () => {
                 const response: protos.google.cloud.iot.v1.IListDeviceConfigVersionsResponse =
                   JSON.parse(data);
+                if (isBinaryDataFormat()) {
+                  response.deviceConfigs?.forEach((element, index) => {
```

Comment:
Use the Array.prototype.map method instead of mutating the elements. Like so:
```
    if (isBinaryDataFormat()) {
      resolve(
        response.deviceConfigs?.map((element) => {
          return {
            ...element,
            cloudUpdateTime: timeSecondsNanos(element.cloudUpdateTime),
            deviceAckTime: timeSecondsNanos(element.deviceAckTime),
            binaryData: new TextEncoder().encode(element.binaryData?.toString()),
          }
        }) ?? [],
      );
      return;
    }
```